client/network-gossip/src/lib.rs

```diff
@@ -97,8 +103,26 @@ impl<B: BlockT, H: ExHashT> Network<B> for Arc<NetworkService<B, H>> {
 		NetworkService::report_peer(self, peer_id, reputation);
 	}
 
-	fn disconnect_peer(&self, who: PeerId) {
-		NetworkService::disconnect_peer(self, who)
+	fn add_set_reserved(&self, who: PeerId, protocol: Cow<'static, str>) {
+		let addr = iter::once(multiaddr::Protocol::P2p(who.into()))
+			.collect::<multiaddr::Multiaddr>();
+		let result = NetworkService::add_to_peers_set(self, protocol, iter::once(addr).collect());
+		if let Err(err) = result {
+			log::error!(target: "gossip", "add_set_reserved failed: {}", err);
+		}
+	}
+
+	fn remove_set_reserved(&self, who: PeerId, protocol: Cow<'static, str>) {
+		let addr = iter::once(multiaddr::Protocol::P2p(who.into()))
+			.collect::<multiaddr::Multiaddr>();
+		let result = NetworkService::remove_from_peers_set(self, protocol, iter::once(addr).collect());
```

Comment:
Same here, should this not call `remove_peers_set_reserved`?

```suggestion
		let result = NetworkService::remove_peers_set_reserved(self, protocol, iter::once(addr).collect());
```